webauthn-security-key/public/auth.client.js

```diff
@@ -0,0 +1,113 @@
+/*
+ * @license
+ * Copyright 2021 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License
+ */
+
+import { decodeServerOptions, encodeCredential } from './encoding.js';
+
+const authStatuses = Object.freeze({
+  NEED_SECOND_FACTOR: 'needSecondFactor',
+  COMPLETE: 'complete',
+});
+
+async function _fetch(path, method, payload = '') {
+  const headers = {
+    'X-Requested-With': 'XMLHttpRequest',
+  };
+  if (payload && !(payload instanceof FormData)) {
+    headers['Content-Type'] = 'application/json';
+    payload = JSON.stringify(payload);
+  }
+  const res = await fetch(path, {
+    method,
+    credentials: 'same-origin',
+    headers: headers,
+    ...(method !== 'GET' && { body: payload }),
+  });
+  if (res.status === 200) {
+    // Server authentication succeeded
+    return res.json();
+  } else {
+    // Server authentication failed
+    const result = await res.json();
+    throw result.error;
+  }
+}
+
+async function registerCredential(name) {
+  const credentialCreationOptionsFromServer = await _fetch(
```

Comment:
@agektmr 
Do we need to add feature detection? If so, how?
Not sure whether this is relevant https://developers.google.com/web/updates/2018/03/webauthn-credential-management ?

Reply:
In my codelab, it's happening inlined with the HTML.
https://glitch.com/edit/#!/webauthn-codelab?path=views%2Fhome.html%3A64%3A1

Reply:
Note: gate cred creation and signin behind feature detection + add a banner in the account UI

Reply:
`if (window.PublicKeyCredential)`